20250226-actors-pubsub.md

```diff
@@ -0,0 +1,158 @@
+# Actor PubSub Implementation Proposal
+
+- Author(s): @joshvanl
+
+## Overview
+
+This proposal introduces PubSub capabilities for Dapr Actors, enabling actors to subscribe to messages for their actor type.
+Additionally, actors of a that type have the ability to publish messages to that type, to be delivered to  particular actor IDs.
+A single designated PubSub component will be marked as the actor PubSub for that namespace, similar to how an actor state store is defined.
+There can be only one actor PubSub Component per namespace.
+Actors will be spawned, if they are not already, when they receive an actor PubSub message.
+
+## Background
+
+Today, Dapr provides PubSub functionality for applications but lacks native PubSub integration for actors.
+This limitation means that actors cannot communicate efficiently via publish-subscribe messaging patterns within their type.
+By enabling Actor PubSub, actors can publish messages to specific actor IDs, improving scalability and communication reliability.
+
+### Considerations
+
+There are a number of scenarios or use cases which could benefit from an "Actor PubSub" feature.
+These include implicitly or explicitly subscribing to topics as an actor, whether publishing messages will spawn actors, if the message is broadcast to a single or all active actors, and if multiple topics can exist within a single actor type domain.
+This proposal describes a single topic per actor type that is implicitly subscribed to by all actors of that type.
+Other scenarios can be considered in future iterations.
+
+## Implementation Details
+
+### Component
+
+A single PubSub component will be designated as the actor PubSub, similar to how an actor state store is configured.
```

Comment:
I'm not a huge fan of this as:
1) I feel it really limits potential pubsub applications for Dapr Actors to inter-type reliable messaging instead of of allowing actors to respond to events happening across the rest of the deployed apps
2) It forces me to use a single pubsub provider for all actors across a solution (requiring that I split up projects to deploy with different components files for each) and potentially running into quota limits (e.g. 5k active connections on Azure Service Bus), which, across multiple actor Apps, might be hit rather quickly.